File: backend/web api/Models/Teacher.cs
public class Teacher
{
    public int ID { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Email { get; set; }
    public string Password { get; set; }
    public string Phone { get; set; }
    public string Avatar { get; set; }
    public int? CourseID { get; set; }
    public Course Course { get; set; }
}

File: learning-platform/src/app/models/teacher.ts
export class Teacher {
  id?: string;
  firstName: string;
  lastName: string;
  email: string;
  password?: string;
  phone: string;
  profileImage?: string;
  
  constructor(data: Partial<Teacher> = {}) {
    this.id = data.id || '';
    this.firstName = data.firstName || '';
    this.lastName = data.lastName || '';
    this.email = data.email || '';
    this.password = data.password || '';
    this.phone = data.phone || '';
    this.profileImage = data.profileImage || '';
  }
}